docs/Security-Focused-Guide-for-AI-Code-Assistant-Instructions.md

```diff
@@ -1,64 +1,181 @@
-# **Security-Focused Guide for AI Code Assistant Instructions**
+# Security-Focused Guide for AI Code Assistant Instructions
 
-AI code assistants can significantly speed up development, but they need guidance to produce **secure** and robust code. This guide explains how to create custom instructions (e.g. [GitHub Copilot instructions file](https://docs.github.com/en/copilot/how-tos/custom-instructions/adding-repository-custom-instructions-for-github-copilot), [Cline instructions file](https://docs.cline.bot/enterprise-solutions/custom-instructions), [Cursor rules](https://docs.cursor.com/context/rules), [Claude markdown](https://docs.anthropic.com/en/docs/claude-code/common-workflows#create-an-effective-claude-md-file), etc.) that ensure the AI assistant accounts for application code security, supply chain safety, and platform or language-specific considerations and to embed a “security conscience” into the tool. In practice, this means fewer vulnerabilities making it into your codebase. Remember that these instructions should be kept concise, specific, and actionable – the goal is to influence the AI’s behaviour without overwhelming it.
+AI code assistants can significantly speed up development, but they need guidance to produce **secure** and robust code. This guide explains how to create custom instructions (e.g. [GitHub Copilot instructions file](https://docs.github.com/en/copilot/how-tos/custom-instructions/adding-repository-custom-instructions-for-github-copilot), [Cline instructions file](https://docs.cline.bot/enterprise-solutions/custom-instructions), [Cursor rules](https://docs.cursor.com/context/rules), [Claude markdown](https://docs.anthropic.com/en/docs/claude-code/common-workflows#create-an-effective-claude-md-file), etc.) that ensure the AI assistant accounts for application code security, supply chain safety, and platform or language-specific considerations and to embed a "security conscience" into the tool. In practice, this means fewer vulnerabilities making it into your codebase. Remember that these instructions should be kept concise, specific, and actionable – the goal is to influence the AI's behaviour without overwhelming it. [[1]](#1)
+
```

Comment:
We haven't done lengthy experiments on this to prove it works - it's just expert opinion. That's okay, but let's make that clear:

```suggestion
These recommendations are based on expert opinion and various recommendations in the literature. We encourage experimentation and feedback to improve these recommendations. We, as an industry, are together learning how to best use these tools.
```